pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -514,6 +522,30 @@ private void addForwardIndex(GenericRow row, int docId, Map<String, Object> dict
 
         numValuesInfo.updateMVEntry(dictIds.length);
       }
+
+      // Update min/max value for columns
+      BaseMutableDictionary dictionary = _dictionaryMap.get(column);
+      if (dictionary != null) {
+        _minValueMap.put(column, dictionary.getMinVal());
```

Comment:
In case of offheap string dictionary it will create a new string everytime from the serialized values. Why nnot just get the minvalue when needed from the dictionary?

Reply:
Make sense, I will change to just store no-dictionary min/max values.